add mtd spi eeprom driver

JokoMLPA opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description

Adding mtd wrapper for SPI-EEPROMs like at25xxx or M95xxx using mtd interface. 
The structure is similar to mtd_flashpage or mtd_sdcard.

### Testing procedure

You can test the driver using RIOT/examples/filesystem (via littlefs). You need to define MTD0 in your board.h and create a mtd_spi_eeprom_t configuration containing the driver parameters, which has to be passed to the mtd layer.

miri64 committed:
make: Allow for true pseudo-submodules

When providing a pseudo-submodule for a module already using the
`SUBMODULES` mechanism to provide submodules, it is not possible to
create a true pseudo-module as submodule (i.e. one without any code on
its own), since the build system currently always expects there to be a
C file `module_submodule.c`.

This removes this requirement.

nmeum committed:
gnrc_pktbuf_malloc: Terminate when fuzzing packet is freed

Since RIOT is an operating system the native binary will never terminate
[0]. The termination condition for fuzzing GNRC is that the packet was
handled by the network stack and therefore freed. If it is never freed
we will deadlock meaning a memory leak was found, afl should be able to
detect this through timeouts.

This is currently only supported for gnrc_pktbuf_malloc since this is
the pktbuf implementation I used for fuzzing. Implementing this in
pktbuf.h is not possible.

[0]: Except NATIVE_AUTO_EXIT is defined, however, even with that define
set RIOT will only terminate when all threads terminated. Unfortunately,
gnrc_udp and other network threads will never terminate.

aabadie committed:
boards/stm32l4: introduce common clock configuration

aabadie committed:
boards/nucleo-l4*: use shared clock configuration

nmeum committed:
makefiles/toolchain: add support for afl

benpicco committed:
drivers: add AT25xxx series EEPROM

This adds a driver for the ST M95xxx series SPI EEPROMs.
The driver has been tested with the M95M01 EEPROM, but should
work with other chips from that family.

SPI-EEPROMs from other vendors from the families AT25xxx, 25AAxxx,
25LCxxx, CAT25xxx & BR25Sxxx should also in the same way.

benpicco committed:
tests: add test for at25xxx EEPROM driver

nmeum committed:
sys/fuzzing: Initialize

This adds a utility module which is used to write applications for
fuzzing RIOT network modules. The module provides a dummy network
interface which is configured with a static IPv6 addresses for modules
which perform operations on the underlying network interface. Besides,
it contains a utility function for transforming data received on
standard input into a `gnrc_pktsnip_t`.

nmeum committed:
fuzzing/gnrc_tcp: Initialize

nmeum committed:
gnrc_tcp: disable checksum checks during fuzzing

aabadie committed:
cpu/fe310: implement driver for watchdog

leandrolanzieri committed:
pkg/wakaama: Move configurations to 'CONFIG_' namespace

Macros that changed:
LWM2M_STANDARD_PORT -> CONFIG_LWM2M_STANDARD_PORT
LWM2M_DTLS_PORT -> CONFIG_LWM2M_DTLS_PORT
LWM2M_BSSERVER_PORT -> CONFIG_LWM2M_BSSERVER_PORT
LWM2M_LOCAL_PORT -> CONFIG_LWM2M_LOCAL_PORT
LWM2M_DEVICE_NAME -> CONFIG_LWM2M_DEVICE_NAME
LWM2M_DEVICE_TTL -> CONFIG_LWM2M_DEVICE_TTL
LWM2M_SERVER_URI -> CONFIG_LWM2M_SERVER_URI
LWM2M_SERVER_ID -> CONFIG_LWM2M_SERVER_ID
LWM2M_ALT_PATH -> CONFIG_LWM2M_ALT_PATH
LWM2M_BOOTSTRAP -> CONFIG_LWM2M_BOOTSTRAP
LWM2M_DEVICE_MANUFACTURER -> CONFIG_LWM2M_DEVICE_MANUFACTURER
LWM2M_DEVICE_MODEL -> CONFIG_LWM2M_DEVICE_MODEL
LWM2M_DEVICE_SERIAL -> CONFIG_LWM2M_DEVICE_SERIAL
LWM2M_DEVICE_FW_VERSION -> CONFIG_LWM2M_DEVICE_FW_VERSION
LWM2M_DEVICE_BINDINGS -> CONFIG_LWM2M_DEVICE_BINDINGS
LWM2M_DEVICE_TYPE -> CONFIG_LWM2M_DEVICE_TYPE
LWM2M_DEVICE_HW_VERSION -> CONFIG_LWM2M_DEVICE_HW_VERSION
LWM2M_DEVICE_SW_VERSION -> CONFIG_LWM2M_DEVICE_SW_VERSION
LWM2M_WITH_LOGS -> CONFIG_LWM2M_WITH_LOGS

leandrolanzieri committed:
Kconfig: Expose pkg/wakaama configurations

leandrolanzieri committed:
examples/wakaama: Configure server only if not set by Kconfig

leandrolanzieri committed:
pkg/wakaama: Move client mode flag to package

As the only implementation available in RIOT of LwM2M is the client, it
makes no sense to set this flag in every application.

nmeum committed:
fuzzing: Initialize

This adds a new subdirectory called `fuzzing/` which will contain
applications for fuzzing various RIOT network modules in the future.
This subdirectory is heavily inspired by the `examples/` subdirectory.

The fuzzing applications use AFL as a fuzzer. Each application contains
Makefiles, source code, and an input corpus used by AFL to generate
input for fuzzing.

nmeum committed:
gnrc_sock: Implement termination condition for fuzzing

The termination condition implemented in gnrc_pktbuf_malloc does not
work when using the sock interface as sock copies packet data to a local
buffer and frees the packet afterwards. As such, the fuzzing application
would exit before performing any input processing.

For this reason, the termination condition in gnrc_pktbuf_malloc is
disabled when using sock. Instead, the application terminates if
gnrc_sock_recv previously returned the fuzzing packet. The underlying
assumption of this implementation is that gnrc_sock_recv is called in a
loop.

aabadie committed:
boards/arduino-nano-33-ble: initial support

kaspar030 committed:
sys/xtimer: introduce xtimer_left_usec()

kaspar030 committed:
sys/evtimer: make use of xtimer_left_usec()

kaspar030 committed:
sys/shell: rpl: make use of xtimer_left_usec()

kaspar030 committed:
tests/rng: remove explicit zeroing of xtimer members

kaspar030 committed:
tests/bench_timers: remove explicit zeroing of xtimer members

kaspar030 committed:
pkg/semtech-loramac: TimerInit(): use static struct initializer

Previously, TimerInit() explicitly set some values of the xtimer truct
to "0". This commit changes it to use a static struct initializer
`(xtimer_t){0}`. The advantage is that now this is independent of the
actual xtimer structure.

Now this also resets the remaining xtimer fields, but they're properly
re-populated elsewhere (by using xtimer_set_msg()).

kaspar030 committed:
pkg/ndn-riot: update patch to do implizit xtimer_t zeroing

leandrolanzieri committed:
net/nanocoap: Move NANOCOAP_NOPTS_MAX to 'CONFIG_' namespace

leandrolanzieri committed:
net/nanocoap: Move NANOCOAP_URI_MAX to 'CONFIG_' namespace

leandrolanzieri committed:
net/nanocoap: Move NANOCOAP_BLOCK_SIZE_EXP_MAX to 'CONFIG_' namespace

leandrolanzieri committed:
net/nanocoap: Move NANOCOAP_QS_MAX to 'CONFIG_' namespace

leandrolanzieri committed:
net/application_layer: Add Kconfig file and CoAP menu

leandrolanzieri committed:
net/nanocoap: Expose configurations to Kconfig

aabadie committed:
makefiles/tools: add support for nrfutil

aabadie committed:
boards/adafruit-clue: add initial support

leandrolanzieri committed:
drivers/mrf24j40: Turn MRF24J40_TEST_SPI_CONNECTION into bool config

MRF24J40_TEST_SPI_CONNECTION is moved to the 'CONFIG_' namespace and by
default is not set. Now it is checked in code using 'IS_ACTIVE'.

leandrolanzieri committed:
drivers/mrf24j40: Turn MRF24J40_USE_EXT_PA_LNA into bool config

MRF24J40_USE_EXT_PA_LNA is moved to the 'CONFIG_' namespace and by
default is not set. Now it is checked in code using 'IS_ACTIVE'.

leandrolanzieri committed:
drivers/mrf24j40: Expose configurations to Kconfig

leandrolanzieri committed:
drivers/Makefile.dep: Set CONFIG_MRF24J40_USE_EXT_PA_LNA if not Kconfig

For the mrf24j40 check if Kconfig is being used before setting the
configuration CFLAG.

jia200x committed:
ieee802154: Expose configurations to Kconfig

jia200x committed:
makefiles/default-radio-settings: disable if Kconfig

If Kconfig is used for configuring default parameters of radios, this
logic has to be disabled in order to avoid collisions of config macros

jia200x committed:
ieee802154: add CONFIG_ prefix to config macros

leandrolanzieri committed:
pkg/wakaama: Move LWM2M_TLSF_BUFFER to 'CONFIG_' namespace

leandrolanzieri committed:
usbus/cdc/acm: Move USBUS_CDC_ACM_STDIO_BUF_SIZE to 'CONFIG_' namespace

leandrolanzieri committed:
usbus/cdc/acm: Move USBUS_CDC_ACM_BULK_EP_SIZE to 'CONFIG_' namespace

leandrolanzieri committed:
usbus/cdc/acm: Expose configurations to Kconfig

leandrolanzieri committed:
makefiles/kconfig: Output out.config with generated configuration

'merged.config' may not always be present (e.g. when no files to merge
are present). In order to always have an up-to-date configuration file
'out.config' will be generated mirroring the content of 'autoconf.h'.
This is the file that the build system will include to read the current
configuration symbols.

leandrolanzieri committed:
Kconfig: Include BOARD and CPU Kconfig files if present

leandrolanzieri committed:
boards/Kconfig: Declare common board symbol

leandrolanzieri committed:
cpu/Kconfig: Declare common CPU symbols

The symbols used to define a CPU are:
- CPU
- CPU_MODEL
- CPU_FAMILY
- CPU_ARCH

aabadie committed:
examples/default: add adafruit-clue to netif boards

fjmolinas committed:
drivers/mtd_spi_nor: power up MTD on init

aabadie committed:
tests: handle cases for bootloader_nrfutil blacklist

aabadie committed:
dist/tools: add bossa version for nrf52

aabadie committed:
makefiles/bossa: only use ROM_OFFSET with version 1.9

aabadie committed:
makefiles/stdio: handle case for bootloader_nrfutil blacklist

akshaim committed:
drivers/include/cc110x.h : Add group

Add compile parameters to config_drivers_netdev group to Doxygen

Co-Authored-By: Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>

jia200x committed:
at86rf2xx: add basic mode PHY state

jia200x committed:
at86rf2xx: implement basic mode

jia200x committed:
at86rf2xx: report CRC error in basic mode

benpicco committed:
cpu/samd21: pwm: fix GCLK_ID & APBCMASK calculation

GCLK_ID and APBCMASK entries are not always uniform.
The previous hack would already break for TCC3.

Just explosively write down the cases, there are only 5 at most.

kb2ma committed:
net/nanocoap: use new Uri-Query functions as primary implementation

kb2ma committed:
net/gcoap: reference canonical function name to add Uri-Query option

kb2ma committed:
net/cord: use canonical function name to add Uri-Query option

kb2ma committed:
tests/nanocoap: use canonical function name to add Uri-Query option

benpicco committed:
cpu/lpc2387: fix check for max number of timers

The CPU has 4 hardware timers.
Configuration for all 4 timers exists, but the compile-time range
check has an off-by-one error, causing the last timer to remain
inaccessible.

cgundogan committed:
Merge pull request #13804 from chrysn-pull-requests/uri-test-isabsolute

uri_parser: Fix and test non-relative recognition

Commit:
tinydtls_sock_dtls: remove duplicate handle message

maribu committed:
Merge pull request #13817 from benpicco/cpu/lpc2387-timer_num

cpu/lpc2387: fix check for max number of timers

mtreffers committed:
drivers/ws281x: fix documentation

extend ws281x documentation with usage for ESP32 and VT100

fix indendation

akshaim committed:
drivers/doc.txt : Add sub-groups

Added Compile Config subgroups for sensors, netdev, actuators,
storage,soft_periph and misc to Doxygen

akshaim committed:
drivers/include/hdc1000.h : Compile Configs

akshaim committed:
drivers/include/ili9341.h : Compile Configs

akshaim committed:
drivers/include/kw2xrf.h : Compile Configs

akshaim committed:
drivers/include/kw41zrf.h : Compile Configs

akshaim committed:
drivers/include/mag3110.h : Compile Configs

akshaim committed:
drivers/include/motor_driver.h : Compile Configs

akshaim committed:
drivers/include/mpl3115a2.h : Compile Configs

akshaim committed:
drivers/include/mtd_sdcard.h : Compile Configs

akshaim committed:
drivers/include/opt3001.h : Compile Configs

akshaim committed:
drivers/include/pir.h : Compile Configs

akshaim committed:
drivers/include/pn532.h : Compile Configs

akshaim committed:
drivers/include/rn2xx3.h : Compile Configs

akshaim committed:
drivers/include/slipdev.h : Compile Configs

akshaim committed:
drivers/include/soft_spi.h : Compile Configs

akshaim committed:
drivers/include/sps30.h : Compile Configs

akshaim committed:
drivers/include/tcs37727.h : Compile Configs

akshaim committed:
drivers/include/tmp00x.h : Compile Configs

akshaim committed:
drivers/include/ethos.h : Compile Configs

fjmolinas committed:
Merge pull request #13794 from aabadie/pr/boards/samd21_bootloader_cleanup

boards/samd21-bootloader*: small cleanup

akshaim committed:
drivers/include/mrf24j40.h : Modify group

Add to config_netdev group to Doxygen

akshaim committed:
drivers/include/sx127x.h  : Add group

Add to config_netdev group to Doxygen

nmeum committed:
fuzzing: Add some documentation on writing fuzzing applications

benpicco committed:
Merge pull request #13821 from pokgak/pr/fix_dups_sock_dtls

tinydtls_sock_dtls: remove duplicate dtls_handle_message

aabadie committed:
examples/default: add arduino-nano-33-ble to netif list

aabadie committed:
makefiles/openocd: define flash targets in a variable

aabadie committed:
boards*: fix position of openocd variable definitions

kaspar030 committed:
Merge pull request #13719 from aabadie/pr/boards/flash_export_remove

boards*: remove remaining uses of export with some openocd related variables

kb2ma committed:
Merge pull request #13815 from kb2ma/nanocoap/add_canonical_uri_query

net/gcoap: add canonical uri query function names

leandrolanzieri committed:
Merge pull request #13720 from aabadie/pr/native_no_export

native: only export NATIVEINCLUDES in vars.inc.mk

aabadie committed:
Merge pull request #11945 from benpicco/EEPROM-m95m01

drivers: add driver for AT25xxx family of EEPROMs

miri64 committed:
examples/gnrc_border_router: use 64-bit prefix with UHCP

miri64 committed:
Merge pull request #13828 from miri64/examples/fix/br-64-bit-prefix-with-UHCP

examples/gnrc_border_router: use 64-bit prefix with UHCP

bergzand committed:
Merge pull request #13375 from leandrolanzieri/pr/kconfig_migrate/sys/usbus_cdc

usbus/cdc/acm: Expose configurations to Kconfig

leandrolanzieri committed:
examples/wakaama: Improve README

fjmolinas committed:
Merge pull request #12974 from leandrolanzieri/pr/kconfig_migrate/wakaama

Kconfig: Expose pkg/wakaama configurations

fjmolinas committed:
Merge pull request #12663 from aabadie/pr/cpu/fe310_watchdog

cpu/fe310: implement driver for watchdog

jeandudey committed:
cc26xx_cc13xx: fix UART1 initialization

Signed-off-by: Jean Pierre Dudey <jeandudey@hotmail.com>

leandrolanzieri committed:
Merge pull request #13714 from aabadie/pr/boards/stm32l4_common_clock

boards/stm32l4: add common clock configuration

kb2ma committed:
net/nanocoap: remove obsolete functions to add Uri-Query option

akshaim committed:
drivers/include/bmx055.h : Expose Parameters

Add ifndef and Expose Compile Parameters to Doxyen

akshaim committed:
drivers/include/ds1307.h : Remove ifndef for I2C

akshaim committed:
drivers/include/hts221.h : Remove ifndef for I2C

akshaim committed:
drivers/include/isl29020.h : Expose Parameters

Add ifndef and Expose Compile Parameters to Doxyen

akshaim committed:
drivers/include/l3g4200d.h : Expose Parameters

Add ifndef and Expose Compile Parameters to Doxyen

akshaim committed:
drivers/include/lpsxxx.h : Expose Parameters

Add ifndef and Expose Compile Parameters to Doxyen

Co-Authored-By: Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>

akshaim committed:
drivers/include/mpl3115a2.h : Remove ifndef for I2C

akshaim committed:
drivers/include/tcs37727.h : Remove ifndef for I2C

Remove ifndef and better formatting for Doxygen

benpicco committed:
sliptty/start_network.sh: don't default to DHCPv6

DHCPv6 is enabled by the `-d` option.
It should not be enabled in absence of this option.

miri64 committed:
Merge pull request #13830 from benpicco/dist/tools/sliptty-fix

sliptty/start_network.sh: don't default to DHCPv6

benpicco committed:
sliptty/start_network.sh: fix with DHCPv6

The `start_network.sh` script creates a `${TUN}` interface, so
the `${TAP}` variable will always be empty.

This means start_dhcpd() will always fail as the DHCPv6 script is lacking
an interface.

miri64 committed:
Merge pull request #13832 from benpicco/dist/tools/sliptty-fix

sliptty/start_network.sh: fix with DHCPv6

benpicco committed:
gnrc_dhcpv6_client: fix for SLIP link-layer

In a point-to-point serial connection, we don't have L2 addresses.
Set the link type & packet length accordingly and we will get a
prefix from KEA.

leandrolanzieri committed:
Merge pull request #13829 from kb2ma/nanocoap/remove_query_add_opts

net/nanocoap: remove obsolete functions for addition of Uri-Query option

aabadie committed:
Merge pull request #13276 from aabadie/pr/boards/adafruit-clue

boards/adafruit-clue: add initial support

miri64 committed:
Merge pull request #13833 from benpicco/dist/tools/sliptty-fix

gnrc_dhcpv6_client: fix for SLIP link-layer

benpicco committed:
pkg/nanopb: extend the documentation

fjmolinas committed:
Merge pull request #13194 from aabadie/pr/board/arduino-nano-33-ble

board/arduino-nano-33-ble: add initial support

leandrolanzieri committed:
Merge pull request #13792 from benpicco/pkg/nanopb-fixes

pkg/nanopb: fix for proto files with includes

bergzand committed:
Merge pull request #13243 from leandrolanzieri/pr/kconfig_migrate/net/nanocoap

net/nanocoap: Expose configurations to Kconfig

bergzand committed:
Merge pull request #13800 from benpicco/pkg/littlefs2-2.2

pkg/littlefs2: bump version to 2.2

miri64 committed:
tests/gnrc_sock_dns: set `GNRC_NETIF_SINGLE`

miri64 committed:
gnrc_rpl_srh: use AsyncSniffer for sniffing

The sniffer in `tests/gnrc_rpl_srh` always had stability issues, since
it uses some badly synchronized threading of mine. Since v2.4.3 scapy
has its own [asynchronous sniffer] (which is already used in
`tests/gnrc_dhcpv6_client_6lbr`), so I ported my sniffer implementation
to use that instead.

[asynchronous sniffer]: https://scapy.readthedocs.io/en/latest/usage.html#asynchronous-sniffing

miri64 committed:
Merge pull request #13835 from miri64/tests/fix/gnrc_netif_single

tests: fix for `GNRC_NETIF_SINGLE`

benpicco committed:
Merge pull request #13801 from miri64/tools/fix/start_network-interface-init

dist/tools/*/start_network.sh: ensure TUN/TAP interface

dylad committed:
Merge pull request #13797 from benpicco/cpu/samd21-pwm

cpu/samd21: pwm: allow to use channels > 3

leandrolanzieri committed:
Merge pull request #13836 from miri64/tests/enh/gnrc_rpl_srh-asyncsniffer

gnrc_rpl_srh: use AsyncSniffer for sniffing

leandrolanzieri committed:
usbus/cdc/ecm: Move USBUS_CDC_ECM_CONFIG_SPEED_DOWNSTREAM to 'CONFIG_'

leandrolanzieri committed:
usbus/cdc/ecm: Move USBUS_CDC_ECM_CONFIG_SPEED_UPSTREAM to 'CONFIG_'

leandrolanzieri committed:
usbus/cdc/ecm: Move USBUS_CDC_ECM_CONFIG_SPEED to 'CONFIG_'

leandrolanzieri committed:
usbus/cdc/ecm: Expose configurations to Kconfig

benpicco committed:
Merge pull request #13716 from benpicco/Makefile.features-cpu

Makefile.features: provide CPU as a feature

leandrolanzieri committed:
examples/usbus_minimal: Do not run Kconfig by default

leandrolanzieri committed:
tests/usbus_cdc_ecm: Check if USB parameters are set via Kconfig

This also avoids running Kconfig by default, given that there is a
Kconfig file in the application folder.

aabadie committed:
Merge pull request #13715 from leandrolanzieri/pr/kconfig/add_cpu_board_common_symbols

Kconfig: Add CPU and Board common symbols

jia200x committed:
Merge pull request #13314 from leandrolanzieri/pr/kconfig_migrate/drivers/mrf24j40

drivers/mrf24j40: Expose configurations to Kconfig

leandrolanzieri committed:
Merge pull request #13315 from jia200x/pr/kconfig/ieee802154

ieee802154: Expose configurations to Kconfig

fjmolinas committed:
Merge pull request #13798 from jia200x/pr/at86rf2x_basic_mode

at86rf2xx: implement basic mode (v2)

maribu committed:
boards/msbiot: Use common stm32 serial/programmer settings

maribu committed:
Merge pull request #13843 from mtreffers/doc-fixes

Doc: fix typo and extend ws281x documentation

maribu committed:
drivers/ws281x: Fix dependencies

arch_esp32 was missing in FEATURES_REQUIRED_ANY for the ws281x, resulting in
ESP32 boards incorrectly being reported as unsupported.

maribu committed:
drivers/ws281x: Extended doc

Point out that automatic backend selection is performed, when no backend is
selected.

JokoMLPA committed:
add mtd spi eeprom driver

benpicco committed:
Makefile.include: add $(MAKEFILEDIR) helper and use it

$(MAKEFILEDIR) will always evaluate to the directory of the Makefile
where it's being called (evaluated).

This construct is used elsewhere in RIOT, make it more accessable
by defining a special variable with it.

benpicco committed:
cpu/kinetis: use $(MAKEFILEDIR)

benpicco committed:
tests: use $(MAKEFILEDIR)

benpicco committed:
Merge pull request #13845 from maribu/ws281x

drivers/ws281x: Fix dependencies & doc

bergzand committed:
littlefs2: bump to version 2.2.1

akshaim committed:
drivers/adcxx1c_params.h : Add Group

Add CT Params to sensors group in Config for Doxygen

akshaim committed:
drivers/include/at.h : Add Group

Add CT Params to misc group in Config for Doxygen

akshaim committed:
drivers/include/at86rf215.h : Add Group

Add CT Params to netdev group in Config for Doxygen

akshaim committed:
cpu/nrf52/include/nrf802154.h : Add Group

Add CT Params to netdev group in Config for Doxygen

fjmolinas committed:
sys/crypto/modes/ccm: accept input_len=0

CCM may be used on messages with no plaintext data.

fjmolinas committed:
tests/sys_crypto/tests-crypto-modes-ccm: add test for input_len=0

bergzand committed:
Merge pull request #13847 from bergzand/pr/littlefs2/bump_2.2.1

littlefs2: bump to version 2.2.1

benpicco commented:
I think you can just make the PR in https://github.com/RIOT-OS/RIOT
We don't have a persistent RIOT fork, the idea is either to try to get things upstream or do them as external modules / boards.

miri64 committed:
xtimer: set now pointer correctly in _update_short_timers() loop

This fixes `xtimer` to use `xtimer_now64()` instead of `xtimer_now()`
for updating the `*now` variable during the iteration in
`_update_short_timers()` function. The same function is used to
initialize `*now` in `_timer_callback()` below.

While using `xtimer_now()` in this iteration step does not hinder the
proper execution of all timers in the short term timers (for those the
`xtimer` module only looks at the `start_time` member, not the
`long_start_time` member) at least for the current long term time window
(I did not test higher cases), it sets the `long_start_time` member to 0
for all timers following in the list of timers after this iteration
step. However, external modules that rely on this to be correct,
e.g. evtimer [1], fail their calculations when trying to compare to
the current value to `xtimer_now64()`.

[1] https://github.com/RIOT-OS/RIOT/blob/11f3d68/sys/evtimer/evtimer.c#L118-L121

Co-Authored-By: Cenk Gündoğan <mail+dev@gundogan.net>

miri64 committed:
tests/xtimer: add regression test for long_start_time update bug

Co-Authored-By: Julian Holzwarth <julian.holzwarth@fu-berlin.de>

miri64 committed:
xtimer_core: piggy-back style fixes

akshaim committed:
drivers/include/dose.h : Add group

Add compile parameters to config_drivers_netdev group to Doxygen

akshaim committed:
drivers/include/ds1307.h : Add group

Add compile parameters to config_drivers_sensors group to Doxygen

akshaim committed:
drivers/include/fxos8700.h : Add group

Add compile parameters to config_drivers_sensors group to Doxygen

iosabi committed:
Allow to define reserved fields in CortexM vector table.

The ARM CortexM vector table has some reserved fields which are used by
some manufacturers to store their custom image information. In
particular, NXP QN908X stores the checksum, Code Read Protection, image
type and boot block pointer in this region.

This patch allows the cpu and board modules to define the value of these
fields at build time by defining a macro.

kaspar030 committed:
Merge pull request #13850 from miri64/xtimer/fix/now32-overflow

xtimer: set now pointer correctly in _update_short_timers() loop

kaspar030 committed:
tests/xtimer_now32_overflow: simplify

benpicco committed:
cpu/lpc2387: gpio: Fix interrupts on PORT2

The calculation of `_state_index` is broken for `port = 2`

    _gpio_isr_map[n + (port<<1)];

Will not yield the right result. As a consequence, IRQs on Port 2
are not working.
The right thing here would be

    _gpio_isr_map[n + (port ? 32 : 0)];

But we might just re-using the `_isr_map_entry()` function.
Also only iterate as many times as there are set interrupt bits.

benpicco committed:
cpu/lpc2387: gpio: Don't discriminate between rising & falling pins

The `test_irq()` function does not discriminate between rising and
falling pins, so there is no need to handle them separately.

benpicco committed:
boards/mcb2388: define flank of the on-board button

Define the button flank for tests that use this define.

akshaim committed:
drivers/include/ads101x.h : Add Group

Add CT Params to sensors group in Config for Doxygen

maribu committed:
Merge pull request #13857 from benpicco/cpu/lpc2387/gpio-fix_port2

cpu/lpc2387: gpio: Fix interrupts on PORT2

akshaim committed:
drivers/mma8x5x.h : Compile Configs

Add CT Params to sensors group in Config for Doxygen

leandrolanzieri committed:
Merge pull request #13848 from fjmolinas/pr_ccm_no_plaintext

 sys/crypto/modes/ccm: handle input_len = 0

fjmolinas committed:
drivers/mtd_spi_nor: return error on invalid manuf id

benpicco committed:
Merge pull request #13844 from maribu/msbiot

boards/msbiot: Use common stm32 serial/programmer settings

akshaim committed:
drivers/hdc1000 : Add CONFIG_

Add CONFIG_ Prefix for configuring the Compile time macros

leandrolanzieri committed:
Merge pull request #13784 from akshaim/drivers_CT_Config

drivers: Add compile configurations groups and add parameters

aabadie committed:
boards/samr3x-xpro: add riotboot feature

akshaim committed:
drivers/hdc1000 : Expose to Kconfig

Expose Configurations to Kconfig

Co-Authored-By: Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>

dylad committed:
Merge pull request #13381 from benpicco/sam0-jlink-fix

makefiles/sam0.inc.mk: fix detection of JLinkExe

benpicco committed:
Merge pull request #13863 from yogo1212/rtc_iff

tm.h: fix typo

benpicco committed:
Merge pull request #13864 from aabadie/pr/boards/samr3x-riotboot

boards/samr3x-xpro: add riotboot feature

benpicco committed:
Merge pull request #13851 from iosabi/vectors_cortexm

Allow to define reserved fields in CortexM vector table.

jeandudey committed:
cc26xx_cc13xx: fix CCFG offset on x2 variants

Signed-off-by: Jean Pierre Dudey <jeandudey@hotmail.com>

jeandudey committed:
cc26x2_cc13x2: fix FCFG1 register offsets

Signed-off-by: Jean Pierre Dudey <jeandudey@hotmail.com>

jeandudey committed:
cc1312-launchpad: add UART1 configuration

Signed-off-by: Jean Pierre Dudey <jeandudey@hotmail.com>

fjmolinas committed:
Merge pull request #10970 from miri64/make/fix/pseudo-submodules

make: Allow for true pseudo-submodules

fjmolinas committed:
Makefile: remove usage of SUBMODULES_NOFORCE

With #10970 only existing *.c files will be added to SRC when using
the SUBMODULES mechanism, so SUBMODULES_NOFORCE (used to filter out
non existing source files) is now redundant so remove the usage.

fjmolinas committed:
doc/creating-modules.md: correct PSEUDOMODULES documentation

leandrolanzieri committed:
Merge pull request #13769 from akshaim/cc110x_CT_Config

drivers/include/cc110x.h : Tagged compile config

benpicco committed:
Merge pull request #13840 from btcven/2020_04_06-uart1

cc26xx_cc13xx: fix UART1 initialization

benpicco committed:
Merge pull request #13867 from btcven/2020_04_14-ccfg-fcfg1

cc26xx_cc13xx: fix CCFG/FCFG1 register offsets

akshaim committed:
drivers/ads101x : Add CONFIG_

Add CONFIG_ Prefix for configuring the Compile time macros

aabadie committed:
boards/cc1312-launchpad: fix typo + reword

akshaim committed:
drivers/ads101x : Expose to Kconfig

Expose Configurations to Kconfig

Co-Authored-By: Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>

aabadie committed:
Merge pull request #13880 from aabadie/pr/boards/cc1312_launchpad_typo

boards/cc1312-launchpad: fix typo + reword

fjmolinas committed:
core/include: add xtstr macro

fjmolinas committed:
cpu/esp_common/include: include common xtstr header

leandrolanzieri committed:
Merge pull request #13870 from akshaim/kconfig_hdc1000_1

drivers/hdc1000 : Expose Configurations to Kconfig

fjmolinas committed:
Merge pull request #13853 from kaspar030/simplify_xtimer_now32_overflow_test

tests/xtimer_now32_overflow: simplify

miri64 committed:
tests/emcute: add small inter-packet gap between server replies

On `samr21-xpro` not all UDP packets were received, leading to the
testing automaton to get into an invalid state and failing. This small
delay between `sendto()` fixes that issue.

miri64 committed:
Merge pull request #13871 from fjmolinas/pr_cleanup_pseudo_submodules

Makefile: remove usage of SUBMODULES_NOFORCE

benpicco committed:
Merge pull request #12415 from maribu/nucleo_board_init

boards/common/nucleo: Marked board_init as weak

miri64 committed:
Merge pull request #13882 from miri64/tests/fix/emcute-samr21-xpro

tests/emcute: add small inter-packet gap between server replies

benpicco committed:
dist/tools: add sliptty to host-tools

benpicco committed:
examples/gnrc_border_router: make SLIP use host-tools

miri64 committed:
Merge pull request #13876 from benpicco/examples/gnrc_border_router_slip_fix

dist/tools: add sliptty to host-tools

benpicco committed:
drivers/at25xxx: return read/written bytes to match MTD API

The MTD API expects read() to return the number of read bytes and
write() to return the number of written bytes.

So change the driver API to match that.

benpicco committed:
tests/driver_at25xxx: adapt to change in driver API

kb2ma committed:
net/nanocoap: validate empty message length

kb2ma committed:
net/gcoap: handle CoAP ping request

kb2ma committed:
net/gcoap: create CoAP ping request

kb2ma committed:
examples/gcoap: add ability to generate CoAP ping request

maribu committed:
Merge pull request #13886 from benpicco/drivers/at25xxx_api_conformity

drivers/at25xxx: return read/written bytes to match MTD API

kb2ma committed:
examples/gcoap: fix reference to variable out of scope

kb2ma committed:
examples/gcoap: fix statement formatting

leandrolanzieri committed:
Merge pull request #13831 from akshaim/I2C_Address_Cleanup

I2C Address Cleanup and Documentation

cgundogan committed:
nanocoap: export coap_opt_get_uint()

leandrolanzieri committed:
Merge pull request #13888 from kb2ma/gcoap_cli/fix_variable_scope

examples/gcoap: fix variable scope

leandrolanzieri committed:
Merge pull request #13881 from akshaim/kconfig_ads101x

drivers/ads101x : Expose Configurations to Kconfig

nmeum committed:
makefiles: Add the fuzzing/ directory to APPLICATION_DIRS

This should ensure that fuzzing applications are build by the CI.

akshaim committed:
drivers/cc110x : Add CONFIG_

Add Add CONFIG_ Prefix for CC110X_DEFAULT_CHANNEL

akshaim committed:
drivers/cc110x : Expose to Kconfig

Expose Configurations to Kconfig

miri64 committed:
Merge pull request #13157 from nmeum/pr/fuzzing_tcp_only

Add AFL-based fuzzing setup for network modules

kb2ma committed:
Merge pull request #13893 from cgundogan/pr/nanocoap-get-opt-uint

nanocoap: export coap_opt_get_uint()

leandrolanzieri committed:
Merge pull request #13897 from akshaim/Kconfig_cc110x

drivers/cc110x : Expose Configurations to Kconfig

Commit:
cpu/cc2538: Prevent underflow of the RX FIFO

Commit:
cpu/cc2538: Check for a minimum length to read from a received frame

Commit:
cpu/cc2538: Check CRC of received message after reading message contents

Commit:
cpu/cc2538: Do not check XREG_RSSISTATbits for RSSI_VALID

Commit:
cpu/cc2538: Flush the RX FIFO (if overflowed) after a receive

fjmolinas committed:
core/doc: fix alignment

fjmolinas committed:
Merge pull request #13906 from MBradbury/pr/cc2538_receive

cpu/cc2538: Prevent underflow of the RX FIFO

kaspar030 committed:
tests/gnrc_ipv6_nib: memory-blacklist msb-430h

leandrolanzieri committed:
Merge pull request #13872 from fjmolinas/pr_pseudomodules_doc

doc/creating-modules.md: correct PSEUDOMODULES documentation

bergzand committed:
Merge pull request #13839 from leandrolanzieri/pr/kconfig_migrate/sys/usbus_ecm

usbus/cdc/ecm: Expose configurations to Kconfig

fjmolinas committed:
Merge pull request #13846 from benpicco/Makefile-THISDIR

Makefile.include: add $(MAKEFILEDIR) helper and use it

bergzand committed:
Merge pull request #13210 from kaspar030/add_xtimer_left

sys/xtimer: remove uses of xtimer internal struct members

leandrolanzieri committed:
Merge pull request #13894 from kb2ma/gcoap/ping

net/gcoap: add CoAP ping request

bergzand committed:
Merge pull request #13558 from fjmolinas/pr_mtd_powerup

drivers/mtd_spi_nor: power up MTD on init

bergzand committed:
Merge pull request #13875 from fjmolinas/pr_core_macros

core/includes: add common macros file

JokoMLPA committed:
change return values

JokoMLPA committed:
drivers/at25xxx add mtd_wrapper as submodule

JokoMLPA committed:
renaming and minor fixes
